test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -935,6 +1182,351 @@ public void TestWriteFunctionObjectToExcel()
+            Assert.AreEqual(5, rowList6[0].Data);
+
+
+        }
+        [Test,Category("Failure")]
+        public void WriteModifyGraph_2()
+        {
+            //incomplete
+            // 1.  Write into  an Existing excel file and Run
+            // 2.  Reduce the array size written to an excel file
+            //3.  Rerun the file 
+            
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteModifyGraph_2.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            ViewModel.HomeSpace.Run();
+            var watch = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.ReadFromFile");
+            var watch2 = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+            var watch3 = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("17bf44dd-0285-496f-a388-58649cadbff8");
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            var list1 = watch.CachedValue.GetElements();
+            Assert.AreEqual(6, list1.Count());
+            Assert.IsTrue(list1[0].IsCollection);
+            var rowList = list1[0].GetElements();
+            Assert.AreEqual(5, rowList[0].Data);
+            rowList = list1[1].GetElements();
+            Assert.AreEqual(6, rowList[0].Data);
+            rowList = list1[2].GetElements();
+            Assert.AreEqual(7, rowList[0].Data);
+            rowList = list1[3].GetElements();
+            Assert.AreEqual(8, rowList[0].Data);
+            rowList = list1[4].GetElements();
+            Assert.AreEqual(9, rowList[0].Data);
+            rowList = list1[5].GetElements();
+            Assert.AreEqual(10, rowList[0].Data);
+
+            // decrease the data written into excel
+
+            ConnectorModel.Make(watch3, watch2, 0, 4);
+            ViewModel.HomeSpace.Run();
+
+            Assert.IsTrue(watch2.CachedValue.IsCollection);
+            var list2 = watch2.CachedValue.GetElements();
+            Assert.AreEqual(6, list2.Count());
+
+            Assert.IsTrue(list2[0].IsCollection);
+            var rowList1 = list2[0].GetElements();
+            Assert.AreEqual(2, rowList1[0].Data);
+            var rowList2 = list2[1].GetElements();
+            Assert.AreEqual(3, rowList2[0].Data);
+            var rowList3 = list2[2].GetElements();
+            Assert.AreEqual(4, rowList3[0].Data);
+            var rowList4 = list2[3].GetElements();
+            Assert.AreEqual(5, rowList4[0].Data);
+            var rowList5 = list2[4].GetElements();
+            Assert.IsNull(rowList5[0].Data);
+            var rowList6 = list2[5].GetElements();
+            Assert.IsNull(rowList6[0].Data);
```

Comment:
the expected output is `{2, 3, 4, 5, 9, 10}` in column form and not `{2, 3, 4, 5, null, null}`

Remove the `Assert.IsNull` statements and replace with:
```
var rowList5 = list2[4].GetElements();
            Assert.AreEqual(9, rowList5[0].Data);
            var rowList6 = list2[5].GetElements();
            Assert.AreEqual(10, rowList6[0].Data);
```